### DISCUSSION TITLE
Some thoughts about the Chat tab (IRC)

### DISCUSSION THREAD

**Discussion by cardpuncher** [ansani/Shareaza#77] [Ideas]
Hi,

It looks like the p2pchat.net domain (which is still registered according to a whois) is refusing connections. I get a "Unable to create a connection to shareaza.p2pchat.net." message with Shareaza. It might make sense to find another IRC network that could accept to host Shareaza's chat channels (irc.p2p-network.net maybe?).

Also, this is secondary but I'm wondering if it would be useful to add the XDCC file downloading feature to Shareaza (with Resume support). Note sure that would target the audience of Shareaza which focuses on ease of use, and another  downside is that files on IRC are often offered as tarballs, the upside is that you don't even need to open ports in your firewall and this could improve file diversity in Shareaza. Just a thought.

**Comment by ansani** [comment]
> Hi,
> 
> 
> 
> It looks like the p2pchat.net domain (which is still registered according to a whois) is refusing connections. I get a "Unable to create a connection to shareaza.p2pchat.net." message with Shareaza. It might make sense to find another IRC network that could accept to host Shareaza's chat channels (irc.p2p-network.net maybe?).
> 
> 
I will work on this topic in the upcoming weeks.

**Comment by ansani** [comment]
> Also, this is secondary but I'm wondering if it would be useful to add the XDCC file downloading feature to Shareaza (with Resume support). Note sure that would target the audience of Shareaza which focuses on ease of use, and another  downside is that files on IRC are often offered as tarballs, the upside is that you don't even need to open ports in your firewall and this could improve file diversity in Shareaza. Just a thought.

I analyzed the XDCC protocol and the possible issue is about the file hash. When I download from ED2K or Gnutella, I can request specific pieces of files using hash and bytes (start and end). With XDCC protocol I cannot check if the requested file is exactly the one I'm expecting to download, as I cannot calculate or request hash 😞

**Comment by cardpuncher** [comment]
On 26.11.2022 14:21, Salvatore Ansani wrote:

> 
> I analyzed the XDCC protocol and the possible issue is about the file 
> hash. When I download from ED2K or Gnutella, I can request specific 
> pieces of files using hash and bytes (start and end). With XDCC protocol 
> I cannot check if the requested file is exactly the one I'm expecting to 
> download, as I cannot calculate or request hash 😞

The original idea wasn't to download a file from IRC *and* from P2P 
networks concurrently. XDCC bots indeed do not support hashes, they work 
only on file name (they're usually well curated), and they tend to offer 
decent download speeds so using a single source shouldn't be much of an 
issue.

In my mind, the idea was that a user would be downloading something from 
IRC, which would then land in his or her downloads folder and only then 
start to be hashed and shared on networks to which Shareaza is connected.

The downside is that users would usually have to know basics about DCC 
file transfers, find a network and channel and know how to request a 
packet. But maybe, just maybe, implementing DCC file transfers and 
leaving it at that, letting the users find their was would be an 
acceptable approach. I'm not even sure that this is a very good idea, 
and it also depends on the feasibility of the implementation. If it's 
easy, or if it's really hard so that's just a suggestion, nothing more.
